gneiss/balances.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import division
+import numpy as np
+from skbio.stats.composition import clr_inv
+from collections import OrderedDict
+
+
+def _balance_basis(tree_node):
+    """ Helper method for calculating balance basis
+    """
+
+    n_tips = sum([n.is_tip() for n in tree_node.traverse()])
```

Comment:
Although, there will be an extra loop required anyways, since `is_tip()` will need to be evaluated over all of the nodes anyways.  So there won't be any savings in this case.  Does that make sense?